pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -660,6 +687,59 @@ const (
 	LazyActivation ActivationPreference = "Lazy"
 )
 
+// SchedulePriority defines how Karmada should resolve the priority and preemption policy
+// for workload scheduling.
+type SchedulePriority struct {
+	// PriorityClassSource specifies where Karmada should look for the PriorityClass definition.
+	// Available options:
+	// - KubePriorityClass (default): Uses Kubernetes PriorityClass (scheduling.k8s.io/v1)
+	// - PodPriorityClass: Uses PriorityClassName from PodTemplate (PodSpec.PriorityClassName)
+	// - FederatedPriorityClass: Uses Karmada FederatedPriorityClass (not yet implemented)
+	//
+	// +kubebuilder:default="KubePriorityClass"
+	// +kubebuilder:validation:Enum=FederatedPriorityClass;KubePriorityClass;PodPriorityClass
+	// +optional
```

Comment:
Currently, we provide support for `KubePriorityClass` only, right?

Shall we add a comment to `PodPriorityClass` to explain a little bit, something like `not yet implemented`?

In addition, please shorten the `+kubebuilder:validation:Enum` with supported options.